cpp/models/lct_secir/initializer_flows.h

```diff
@@ -0,0 +1,251 @@
+                                                         1 / m_model.parameters.template get<TimeExposed>());
+        //C
+        init.segment(LctState::template get_first_index<InfectionState::InfectedNoSymptoms>(),
+                     LctState::template get_num_subcompartments<InfectionState::InfectedNoSymptoms>()) =
+            compute_compartment<InfectionState::InfectedNoSymptoms>(
+                Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms),
+                1 / m_model.parameters.template get<TimeInfectedNoSymptoms>());
+        //I
+        init.segment(LctState::template get_first_index<InfectionState::InfectedSymptoms>(),
+                     LctState::template get_num_subcompartments<InfectionState::InfectedSymptoms>()) =
+            compute_compartment<InfectionState::InfectedSymptoms>(
+                Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms),
+                1 / m_model.parameters.template get<TimeInfectedSymptoms>());
+        //H
+        init.segment(LctState::template get_first_index<InfectionState::InfectedSevere>(),
+                     LctState::template get_num_subcompartments<InfectionState::InfectedSevere>()) =
+            compute_compartment<InfectionState::InfectedSevere>(
+                Eigen::Index(InfectionTransition::InfectedSymptomsToInfectedSevere),
+                1 / m_model.parameters.template get<TimeInfectedSevere>());
+        //U
+        init.segment(LctState::template get_first_index<InfectionState::InfectedCritical>(),
+                     LctState::template get_num_subcompartments<InfectionState::InfectedCritical>()) =
+            compute_compartment<InfectionState::InfectedCritical>(
+                Eigen::Index(InfectionTransition::InfectedSevereToInfectedCritical),
+                1 / m_model.parameters.template get<TimeInfectedCritical>());
+        //R
+        // Number of recovered is equal to the cumulative number of confirmed cases minus the number of people who are infected at the moment.
+        init[LctState::Count - 2] =
+            total_confirmed_cases -
+            init.segment(LctState::template get_first_index<InfectionState::InfectedSymptoms>(),
+                         LctState::template get_num_subcompartments<InfectionState::InfectedSymptoms>() +
+                             LctState::template get_num_subcompartments<InfectionState::InfectedSevere>() +
+                             LctState::template get_num_subcompartments<InfectionState::InfectedCritical>())
+                .sum() -
+            deaths;
+
+        //S
+        init[0] = total_population - init.segment(1, LctState::Count - 2).sum() - deaths;
+
+        //D
+        init[LctState::Count - 1] = deaths;
+
+        for (int i = 0; i < LctState::Count; i++) {
+            if (init[i] < 0) {
+                log_error("Initialization failed. Not enough time points for the transitions are given. More than {} "
```

Comment:
Is it possible that the size of the subcompartments negative although there were enough time points, for example if there are negative flows given?

Reply:
Yes it is, thanks, I adapted this. I think it is okay now.